### ISSUE TITLE
entk version 0.7.4

### ISSUE THREAD

**Issue by euhruska** [radical-collaboration/extasy-grlsd#52]
I did a fresh installation of extasy for cpu and entk was missing from the stack:
```

  python               : 2.7.14
  pythonpath           :
  virtualenv           : extasy6

  radical.analytics    : v0.45.2-102-gaec2e1d@devel
  radical.pilot        : 0.47.12-v0.47.12-167-ge60920ef@devel
  radical.utils        : 0.47.4-merge-pre_gpu-22-ga942c4b@devel
  saga                 : 0.47.4-v0.47.4-32-g71a97659@devel
```

But when I try to install entk it claims the requirement is already sattisfied:
```
pip install git+https://github.com/radical-cybertools/radical.entk.git@devel
Collecting git+https://github.com/radical-cybertools/radical.entk.git@devel
  Cloning https://github.com/radical-cybertools/radical.entk.git (to devel) to /tmp/pip-21QLvq-build
  Requirement already satisfied (use --upgrade to upgrade): radical.entk==0.6.2 from git+https://github.com/radical-cybertools/radical.entk.git@devel in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages
Requirement already satisfied: radical.utils in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.entk==0.6.2)
Requirement already satisfied: pika in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.entk==0.6.2)
Requirement already satisfied: radical.pilot in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.entk==0.6.2)
Requirement already satisfied: autodoc in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.entk==0.6.2)
Requirement already satisfied: pytest in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.entk==0.6.2)
Requirement already satisfied: hypothesis in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.entk==0.6.2)
Requirement already satisfied: sphinx in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.entk==0.6.2)
Requirement already satisfied: future in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.utils->radical.entk==0.6.2)
Requirement already satisfied: colorama in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.utils->radical.entk==0.6.2)
Requirement already satisfied: netifaces==0.10.4 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.utils->radical.entk==0.6.2)
Requirement already satisfied: setproctitle in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.utils->radical.entk==0.6.2)
Requirement already satisfied: saga-python>=0.44 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: pymongo==2.8 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: python-hostlist in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: ntplib in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: msgpack-python in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: pyzmq in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: webtest in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from autodoc->radical.entk==0.6.2)
Requirement already satisfied: decorator in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from autodoc->radical.entk==0.6.2)
Requirement already satisfied: pluggy<0.7,>=0.5 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: atomicwrites>=1.0 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: more-itertools>=4.0.0 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: py>=1.5.0 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: attrs>=17.4.0 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: funcsigs; python_version < "3.0" in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: setuptools in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: six>=1.10.0 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from pytest->radical.entk==0.6.2)
Requirement already satisfied: enum34; python_version == "2.7" in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from hypothesis->radical.entk==0.6.2)
Requirement already satisfied: coverage in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from hypothesis->radical.entk==0.6.2)
Requirement already satisfied: imagesize in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: Pygments>=2.0 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: sphinxcontrib-websupport in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: alabaster<0.8,>=0.7 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: snowballstemmer>=1.1 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: docutils>=0.11 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: typing; python_version < "3.5" in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: babel!=2.0,>=1.3 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: requests>=2.0.0 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: packaging in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: Jinja2>=2.3 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from sphinx->radical.entk==0.6.2)
Requirement already satisfied: apache-libcloud in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from saga-python>=0.44->radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: parse in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from saga-python>=0.44->radical.pilot->radical.entk==0.6.2)
Requirement already satisfied: beautifulsoup4 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from webtest->autodoc->radical.entk==0.6.2)
Requirement already satisfied: waitress>=0.8.5 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from webtest->autodoc->radical.entk==0.6.2)
Requirement already satisfied: WebOb>=1.2 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from webtest->autodoc->radical.entk==0.6.2)
Requirement already satisfied: pytz>=0a in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from babel!=2.0,>=1.3->sphinx->radical.entk==0.6.2)
Requirement already satisfied: certifi>=2017.4.17 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from requests>=2.0.0->sphinx->radical.entk==0.6.2)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from requests>=2.0.0->sphinx->radical.entk==0.6.2)
Requirement already satisfied: idna<2.7,>=2.5 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from requests>=2.0.0->sphinx->radical.entk==0.6.2)
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from requests>=2.0.0->sphinx->radical.entk==0.6.2)
Requirement already satisfied: pyparsing>=2.0.2 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from packaging->sphinx->radical.entk==0.6.2)
Requirement already satisfied: MarkupSafe>=0.23 in /scratch1/eh22/conda/envs/extasy6/lib/python2.7/site-packages (from Jinja2>=2.3->sphinx->radical.entk==0.6.2)
You are using pip version 9.0.3, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```
How can I force it to install or recognize?

**Comment by vivek-bala** [comment]
You can do it by ```pip install <package name> --upgrade``` inside your virtualenv.

**Comment by euhruska** [comment]
now I get at radical-stack of
```
  python               : 2.7.14
  pythonpath           :
  virtualenv           : extasy6

  radical.analytics    : v0.45.2-102-gaec2e1d@devel
  radical.entk         : 0.6.2-0.6.2-2-g6526f23@HEAD-detached-at-6526f23
  radical.pilot        : 0.47.13
  radical.utils        : 0.47.5
  saga                 : 0.47.6
```
is that correct for running extasy on bw (only cpu)?

**Comment by euhruska** [comment]
With the above stack I get a bootstrap error similar to the gpu version, but here I used only devel branches
```
# unpacking virtualenv tgz
# cmd: tar zxmf 'virtualenv-1.9.tar.gz'
#
#
# SUCCESS
#
# -------------------------------------------------------------------

# -------------------------------------------------------------------
#
# Create virtualenv
# cmd: /sw/bw/bwpy/mnt/bin/python virtualenv-1.9/virtualenv.py /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.47.13
#
Failed to import the site module
Traceback (most recent call last):
  File "/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.47.13/lib/python3.5/site.py", line 67, in <module>
    import os
  File "/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.47.13/lib/python3.5/os.py", line 708, in <module>
    from _collections_abc import MutableMapping
ImportError: No module named '_collections_abc'
Using base prefix '/mnt/bwpy/single/usr/lib/python-exec/python3.5/../../..'
New python executable in /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.47.13/bin/python
ERROR: The executable /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.47.13/bin/python is not functioning
ERROR: It thinks sys.prefix is '/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/rp.session.leonardo.rice.edu.eh22.017687.0004/pilot.0000' (should be '/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.47.13')
ERROR: virtualenv is not compatible with this system or executable
#
# ERROR
# no fallback command available
#
# -------------------------------------------------------------------
ERROR: Couldn't create virtualenv
Error on virtenv creation -- abort
removed `/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.47.13.lock'
```

**Comment by andre-merzky** [comment]
The `--upgrade` has pulled the released versions of RP, whereas the initial installation showed devel branches.  I am uncertain which set of branches is required by extacy.  At the same time, I don't seem to recollect any problems which relate to `_collections_abc` import, so maybe this is a new one.   But before digging into it, @vivek-bala , can you please clarify the required RCT stack version?

**Comment by vivek-bala** [comment]
For the cpu version of extasy-grlsd (where you don't cpu_reqs and gpu_reqs attributes for a Task), please use the master branch of EnTK and RP. When using the gpu version of the extasy-grlsd, you should use the feature/gpu version in EnTK and devel of RP (based on #53).

**Comment by euhruska** [comment]
ok, but that didn't solve the bootstrap issue

**Comment by euhruska** [comment]
waiting on https://github.com/radical-collaboration/extasy-grlsd/issues/53, since similar issue

**Comment by euhruska** [comment]
while reinstalling the feature/gpu branch of entk doesn't exist, but using devel I get
```ImportError: cannot import name ResourceManager```
Which branch should I use?

**Comment by andre-merzky** [comment]
> @shantenujha: CC-ing Andre explicitly, as I’m not sure Andre monitors / gets this git repo.

I do watch - but alas I do not know the answer, I am afraid @vivek-bala has to answer for EnTK branches...

**Comment by vivek-bala** [comment]
Hey Eugen,

There were some API changes with the latest release. You should have
received them an email earlier and
they are also documented in
https://github.com/radical-cybertools/radical.entk/blob/master/changes.MD.

Let me know how it goes!

Thanks,
Vivek

**Comment by euhruska** [comment]
Tried to update to entk 0.7.0 in script at https://github.com/radical-collaboration/extasy-grlsd/blob/master/extasy_grlsd_2.py, but go an error.  wf is an Pipeline() object, but how do I make it iterable?
```
Error: 'Pipeline' object is not iterable
Traceback (most recent call last):
  File "extasy_grlsd_2.py", line 363, in <module>
    appman.workflow = wf
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/appman/appmanager.py", line 206, in workflow
    for p in workflow:
TypeError: 'Pipeline' object is not iterable
```

radical-stack
```
  python               : 2.7.14
  pythonpath           :
  virtualenv           : extasy10

  radical.analytics    : v0.45.2-102-gaec2e1d@devel
  radical.entk         : 0.7.4-0.7.3-11-g4d95fcb7@devel
  radical.pilot        : 0.50.7-v0.50.7-5-gc1d54e95@devel
  radical.utils        : 0.50.1-v0.50.1-1-gce60836@devel
  saga                 : 0.50.0-v0.50.0@devel
```

**Comment by vivek-bala** [comment]
Hey Eugen, can you set ```appman.workflow = set(wf)``` in line https://github.com/radical-collaboration/extasy-grlsd/blob/master/extasy_grlsd_2.py#L363. The workflow needs to be a set of pipelines.

**Comment by euhruska** [comment]
No, same error
```
Error: 'Pipeline' object is not iterable
Traceback (most recent call last):
  File "extasy_grlsd_2.py", line 363, in <module>
    appman.workflow = set(wf)
TypeError: 'Pipeline' object is not iterable
```

**Comment by vivek-bala** [comment]
I don't think that got converted to a set. Can you try ```appman.workflow = set([wf])```? ```appman.workflow = [wf]``` should also work, but list order is not guaranteed.

**Comment by euhruska** [comment]
set([wf]) worked, but got stuck in boostrap:
```
# SUCCESS
#
# -------------------------------------------------------------------
6.7102,ve_activate_start,bootstrap_0,MainThread,pilot.0000,PMGR_ACTIVE_PENDING,
PYTHON: /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python
PIP   : /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/pip --cert /scratch/sciteam/hruska/radical.pilot.sandbox/rp.session.leonardo.rice.edu.eh22.017749.0004/pilot.0000/../cacert.pem
/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory
/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory
PYTHON INTERPRETER: /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python
PYTHON_VERSION    :
VE_MOD_PREFIX     :
PIP installer     : /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/pip --cert /scratch/sciteam/hruska/radical.pilot.sandbox/rp.session.leonardo.rice.edu.eh22.017749.0004/pilot.0000/../cacert.pem
/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python2.7: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory
PIP version       :
activated virtenv
VIRTENV      : /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7
VE_MOD_PREFIX: ///////
RP_MOD_PREFIX: ///////
PYTHONPATH   : ///////:/opt/xalt/0.7.6/sles11.3/libexec:/opt/cray/sdb/1.1-1.0502.63652.4.27.gem/lib64/py
6.7763,ve_activate_stop,bootstrap_0,MainThread,pilot.0000,PMGR_ACTIVE_PENDING,

# -------------------------------------------------------------------
#
# update setuptools
# cmd: /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/pip install --upgrade setuptools==0.6c11
#
/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python2.7: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory
#
# ERROR
# no fallback command available
#
# -------------------------------------------------------------------
Couldn't update setuptools -- using default version

# -------------------------------------------------------------------
#
# update pip
# cmd: /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/pip install --upgrade pip==1.4.1
#
/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python2.7: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory
#
# ERROR
# no fallback command available
#
# -------------------------------------------------------------------
Couldn't update pip -- using default version
PYTHON: /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python
PIP   : /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/pip --cert /scratch/sciteam/hruska/radical.pilot.sandbox/rp.session.leonardo.rice.edu.eh22.017749.0004/pilot.0000/../cacert.pem

# -------------------------------------------------------------------
#
# install apache-libcloud
# cmd: easy_install --upgrade apache-libcloud
#
/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python2.7: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory
#
# ERROR
# no fallback command available
#
# -------------------------------------------------------------------
Couldn't install/upgrade apache-libcloud! Lets see how far we get ...

# -------------------------------------------------------------------
#
# install pymongo==2.8
# cmd: /mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/pip --cert /scratch/sciteam/hruska/radical.pilot.sandbox/rp.session.leonardo.rice.edu.eh22.017749.0004/pilot.0000/../cacert.pem install  pymongo==2.8
#
/mnt/c/scratch/sciteam/hruska/radical.pilot.sandbox/ve.ncsa.bw_aprun.0.50.7/bin/python2.7: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory
#
# ERROR
# no fallback command available
```

**Comment by vivek-bala** [comment]
Do you have a VE called "ve.ncsa.bw_aprun.0.50.7" in your sandbox? If not can you do the following:
* Inside the sandbox, download script using ```wget https://raw.githubusercontent.com/radical-cybertools/radical.pilot/devel/bin/radical-pilot-create-static-ve```
* create VE by executing "sh radical-pilot-create-static-ve ve.ncsa.bw_aprun.0.50.7 bw"

Then please retry and let me know how it goes.

**Comment by euhruska** [comment]
it automatically created ve.ncsa.bw_aprun.0.50.7 in my sandbox, I removed and reinstalled it, boostrap didn't show an error, but got this
```
elf._handle_read()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 464, in _handle_read
    self._on_data_available(data)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2021, in _on_data_available
    self._process_frame(frame_value)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2142, in _process_frame
    if self._process_callbacks(frame_value):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2123, in _process_callbacks
    frame_value)  # Args
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 60, in wrapper
    return function(*tuple(args), **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 92, in wrapper
    return function(*args, **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 236, in process
    callback(*args, **keywords)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1358, in _on_channel_closed
    method.reply_text)
ChannelClosed: (404, "NOT_FOUND - no queue 're.session.leonardo.rice.edu.eh22.017749.0005-hb-response' in vhost '/'")

2018-08-06 18:40:30,616: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Terminating heartbeat thread
2018-08-06 18:40:30,616: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Hearbeat thread terminated
2018-08-06 18:40:31,882: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Terminating tmgr process
2018-08-06 18:40:31,882: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Task manager process closed
2018-08-06 18:40:31,882: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Restarting task manager process
2018-08-06 18:40:31,883: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Starting task manager process
2018-08-06 18:40:31,885: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Restarting heartbeat thread
2018-08-06 18:40:31,886: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Starting heartbeat thread
2018-08-06 18:40:31,962: radical.entk.task_manager.0000: task-manager                    : MainThread     : INFO    : Task Manager process started
2018-08-06 18:40:32,577: radical.entk.task_manager.0000: MainProcess                     : heartbeat      : INFO    : Sent heartbeat request
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/rp/task_manager.py", line 177, in _tmgr
    umgr = rp.UnitManager(session=rmgr._session)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/pilot/unit_manager.py", line 148, in __init__
    self._session._register_umgr(self)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/pilot/session.py", line 781, in _register_umgr
    self._dbs.insert_umgr(umgr.as_dict())
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/pilot/db/database.py", line 354, in insert_umgr
    result = self._c.insert(umgr_doc)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pymongo/collection.py", line 410, in insert
    _check_write_command_response(results)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pymongo/helpers.py", line 202, in _check_write_command_response
    raise DuplicateKeyError(error.get("errmsg"), 11000, error)
DuplicateKeyError: E11000 duplicate key error index: clementigroup.rp.session.leonardo.rice.edu.eh22.017749.0006.$_id_ dup key: { : "umgr.0000" }

Process task-manager:
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self.run()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/rp/task_manager.py", line 271, in _tmgr
    raise Error(text=ex)
NameError: global name 'Error' is not defined
2018-08-06 18:41:02,714: radical.entk.task_manager.0000: MainProcess                     : heartbeat      : ERROR   : Heartbeat failed with error: (404, "NOT_FOUND - no queue 're.session.leonardo.rice.edu.eh22.017749.0005-hb-response' in vhost '/'")
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/base/task_manager.py", line 151, in _heartbeat
    method_frame, props, body = mq_channel.basic_get(queue=self._hb_response_q)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 2077, in basic_get
    self._basic_getempty_result.is_ready)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1292, in _flush_output
    *waiters)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 458, in _flush_output
    self._impl.ioloop.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 495, in poll
    self._poller.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 1114, in poll
    self._dispatch_fd_events(fd_event_map)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 831, in _dispatch_fd_events
    handler(fileno, events)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 410, in _handle_events
    self._handle_read()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 464, in _handle_read
    self._on_data_available(data)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2021, in _on_data_available
    self._process_frame(frame_value)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2142, in _process_frame
    if self._process_callbacks(frame_value):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2123, in _process_callbacks
    frame_value)  # Args
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 60, in wrapper
    return function(*tuple(args), **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 92, in wrapper
    return function(*args, **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 236, in process
    callback(*args, **keywords)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1358, in _on_channel_closed
    method.reply_text)
ChannelClosed: (404, "NOT_FOUND - no queue 're.session.leonardo.rice.edu.eh22.017749.0005-hb-response' in vhost '/'")
Exception in thread heartbeat:
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/base/task_manager.py", line 151, in _heartbeat
    method_frame, props, body = mq_channel.basic_get(queue=self._hb_response_q)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 2077, in basic_get
    self._basic_getempty_result.is_ready)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1292, in _flush_output
    *waiters)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 458, in _flush_output
    self._impl.ioloop.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 495, in poll
    self._poller.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 1114, in poll
    self._dispatch_fd_events(fd_event_map)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 831, in _dispatch_fd_events
    handler(fileno, events)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 410, in _handle_events
    self._handle_read()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 464, in _handle_read
    self._on_data_available(data)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2021, in _on_data_available
    self._process_frame(frame_value)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2142, in _process_frame
    if self._process_callbacks(frame_value):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2123, in _process_callbacks
    frame_value)  # Args
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 60, in wrapper
    return function(*tuple(args), **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 92, in wrapper
    return function(*args, **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 236, in process
    callback(*args, **keywords)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1358, in _on_channel_closed
    method.reply_text)
ChannelClosed: (404, "NOT_FOUND - no queue 're.session.leonardo.rice.edu.eh22.017749.0005-hb-response' in vhost '/'")

2018-08-06 18:41:02,855: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Terminating heartbeat thread
2018-08-06 18:41:02,855: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Hearbeat thread terminated
2018-08-06 18:41:04,125: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Terminating tmgr process
2018-08-06 18:41:04,125: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Task manager process closed
2018-08-06 18:41:04,125: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Restarting task manager process
2018-08-06 18:41:04,126: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Starting task manager process
2018-08-06 18:41:04,128: radical.entk.appmanager.0000: MainProcess                     : MainThread     : INFO    : Restarting heartbeat thread
2018-08-06 18:41:04,128: radical.entk.task_manager.0000: MainProcess                     : MainThread     : INFO    : Starting heartbeat thread
2018-08-06 18:41:04,191: radical.entk.task_manager.0000: task-manager                    : MainThread     : INFO    : Task Manager process started
2018-08-06 18:41:04,851: radical.entk.task_manager.0000: MainProcess                     : heartbeat      : INFO    : Sent heartbeat request
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/rp/task_manager.py", line 177, in _tmgr
    umgr = rp.UnitManager(session=rmgr._session)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/pilot/unit_manager.py", line 148, in __init__
    self._session._register_umgr(self)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/pilot/session.py", line 781, in _register_umgr
    self._dbs.insert_umgr(umgr.as_dict())
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/pilot/db/database.py", line 354, in insert_umgr
    result = self._c.insert(umgr_doc)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pymongo/collection.py", line 410, in insert
    _check_write_command_response(results)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pymongo/helpers.py", line 202, in _check_write_command_response
    raise DuplicateKeyError(error.get("errmsg"), 11000, error)
DuplicateKeyError: E11000 duplicate key error index: clementigroup.rp.session.leonardo.rice.edu.eh22.017749.0006.$_id_ dup key: { : "umgr.0000" }

Process task-manager:
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self.run()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/rp/task_manager.py", line 271, in _tmgr
    raise Error(text=ex)
NameError: global name 'Error' is not defined
2018-08-06 18:41:34,995: radical.entk.task_manager.0000: MainProcess                     : heartbeat      : ERROR   : Heartbeat failed with error: (404, "NOT_FOUND - no queue 're.session.leonardo.rice.edu.eh22.017749.0005-hb-response' in vhost '/'")
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/base/task_manager.py", line 151, in _heartbeat
    method_frame, props, body = mq_channel.basic_get(queue=self._hb_response_q)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 2077, in basic_get
    self._basic_getempty_result.is_ready)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1292, in _flush_output
    *waiters)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 458, in _flush_output
    self._impl.ioloop.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 495, in poll
    self._poller.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 1114, in poll
    self._dispatch_fd_events(fd_event_map)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 831, in _dispatch_fd_events
    handler(fileno, events)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 410, in _handle_events
    self._handle_read()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 464, in _handle_read
    self._on_data_available(data)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2021, in _on_data_available
    self._process_frame(frame_value)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2142, in _process_frame
    if self._process_callbacks(frame_value):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2123, in _process_callbacks
    frame_value)  # Args
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 60, in wrapper
    return function(*tuple(args), **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 92, in wrapper
    return function(*args, **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 236, in process
    callback(*args, **keywords)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1358, in _on_channel_closed
    method.reply_text)
ChannelClosed: (404, "NOT_FOUND - no queue 're.session.leonardo.rice.edu.eh22.017749.0005-hb-response' in vhost '/'")
Exception in thread heartbeat:
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/radical/entk/execman/base/task_manager.py", line 151, in _heartbeat
    method_frame, props, body = mq_channel.basic_get(queue=self._hb_response_q)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 2077, in basic_get
    self._basic_getempty_result.is_ready)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1292, in _flush_output
    *waiters)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 458, in _flush_output
    self._impl.ioloop.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 495, in poll
    self._poller.poll()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 1114, in poll
    self._dispatch_fd_events(fd_event_map)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 831, in _dispatch_fd_events
    handler(fileno, events)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 410, in _handle_events
    self._handle_read()
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/base_connection.py", line 464, in _handle_read
    self._on_data_available(data)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2021, in _on_data_available
    self._process_frame(frame_value)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2142, in _process_frame
    if self._process_callbacks(frame_value):
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/connection.py", line 2123, in _process_callbacks
    frame_value)  # Args
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 60, in wrapper
    return function(*tuple(args), **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 92, in wrapper
    return function(*args, **kwargs)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/callback.py", line 236, in process
    callback(*args, **keywords)
  File "/scratch1/eh22/conda/envs/extasy10/lib/python2.7/site-packages/pika/adapters/blocking_connection.py", line 1358, in _on_channel_closed
    method.reply_text)
ChannelClosed: (404, "NOT_FOUND - no queue 're.session.leonardo.rice.edu.eh22.017749.0005-hb-response' in vhost '/'")
```

**Comment by vivek-bala** [comment]
Can you attach the entire set of EnTK logs please? I think the "pika channel closed" error is part of the reattempt to restart from the beginning, we need to see why every attempt is failing.

**Comment by euhruska** [comment]
[rp.session.leonardo.rice.edu.eh22.017749.0006.zip](https://github.com/radical-collaboration/extasy-grlsd/files/2267373/rp.session.leonardo.rice.edu.eh22.017749.0006.zip)
[re.session.leonardo.rice.edu.eh22.017749.0005.zip](https://github.com/radical-collaboration/extasy-grlsd/files/2267374/re.session.leonardo.rice.edu.eh22.017749.0005.zip)

are these the correct logs?

**Comment by vivek-bala** [comment]
Hey Eugen, apologies on the delay. Some of the RP logs seem to be empty, can you ensure that you set RADICAL_PILOT_VERBOSE=DEBUG ? Also, the EnTK logs are basically those printed on to the standard when you run your EnTK script once you have set RADICAL_ENTK_VERBOSE=INFO.

You can capture the logs of the standard to a file by doing something like [this](https://stackoverflow.com/questions/418896/how-to-redirect-output-to-a-file-and-stdout#418899)

**Comment by vivek-bala** [comment]
The logs currently present do not indicate the source of the error.

**Comment by vivek-bala** [comment]
Also can you mention the hostname and the port for your RabbitMQ server?

**Comment by euhruska** [comment]
you are right, I didn't set RADICAL_PILOT_VERBOSE at all, rerunning
hostname: leonardo.rice.edu
how do I figure out the port for my RabbitMQ server?

**Comment by vivek-bala** [comment]
If you haven't explicitly set it, it should be 5672. Are you running a docker container? (in which case it might not be 5672).

You can do ```sudo lsof -i | grep beam``` to list and grep for the port number.

**Comment by euhruska** [comment]
no docker container, sudo not allowed
but I got 
```
>lsof -i | grep beam
beam.smp  12064 eh22   59u  IPv4 5048780      0t0  TCP *:25672 (LISTEN)
beam.smp  12064 eh22   60u  IPv4 5048782      0t0  TCP localhost:42598->localhost:epmd (ESTABLISHED)
beam.smp  12064 eh22   61u  IPv4 5048784      0t0  UDP *:53103
beam.smp  12064 eh22   70u  IPv6 5054841      0t0  TCP *:amqp (LISTEN)
```
more logs with entk0.7.5:
```radical-stack
  python               : 2.7.14
  pythonpath           :
  virtualenv           : extasy9

  radical.analytics    : v0.45.2-102-gaec2e1d@devel
  radical.entk         : 0.7.5-0.7.3-19-g383ebea1@devel
  radical.pilot        : 0.50.8-v0.50.8-6-g6e36785f@devel
  radical.utils        : 0.50.1-v0.50.1-1-gce60836@devel
  saga                 : 0.50.0-v0.50.0@devel
```
[entk.log](https://github.com/radical-collaboration/extasy-grlsd/files/2284112/entk.log)
[re.session.giotto.rice.edu.eh22.017756.0007.zip](https://github.com/radical-collaboration/extasy-grlsd/files/2284113/re.session.giotto.rice.edu.eh22.017756.0007.zip)
[rp.session.giotto.rice.edu.eh22.017756.0008.zip](https://github.com/radical-collaboration/extasy-grlsd/files/2284114/rp.session.giotto.rice.edu.eh22.017756.0008.zip)
[rs.log](https://github.com/radical-collaboration/extasy-grlsd/files/2284115/rs.log)

**Comment by vivek-bala** [comment]
Thanks for the entk logs. I think I understand the issue with devel. Following up on the issue here and another ticket, I will suggest using the latest version of EnTK (either master or devel, and release for RP, RS and RU).

Can you remind why you can't use the released version of EnTK 0.7.4? I will place the fix for devel that you are using now (0.7.5).

**Comment by euhruska** [comment]
when I try using entk 0.7.4 or 0.7.5 I get a fatal error around the time when the first unit is created, see the above entk.log 
including 
```2018-08-13 12:38:22,238: radical.entk.task_processor: task-manager                    : MainThread     : ERROR   : CU creation failed, error: attribute set is not extensible/private (key lfs_per_process) (/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/saga/attributes.py +1068 (_attributes_i_set)  :  raise se.IncorrectState ("attribute set is not extensible/private (key %s)" %  key))```
```
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy9/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self.run()
  File "/scratch1/eh22/conda/envs/extasy9/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/radical/entk/execman/rp/task_manager.py", line 271, in _tmgr
    raise Error(text=ex)
NameError: global name 'Error' is not defined
Traceback (most recent call last):
  File "/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/radical/entk/execman/rp/task_manager.py", line 209, in _tmgr
    bulk_cuds.append(create_cud_from_task(t, placeholder_dict, local_prof))
  File "/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/radical/entk/execman/rp/task_processor.py", line 289, in create_cud_from_task
    cud.lfs_per_process = task.lfs_per_process
  File "/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/saga/attributes.py", line 2643, in __setattr__
    return self._attributes_i_set      (key, val, flow=self._DOWN)
  File "/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/saga/attributes.py", line 1068, in _attributes_i_set
    raise se.IncorrectState ("attribute set is not extensible/private (key %s)" %  key)
IncorrectState: attribute set is not extensible/private (key lfs_per_process) (/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/saga/attributes.py +1068 (_attributes_i_set)  :  raise se.IncorrectState ("attribute set is not extensible/private (key %s)" %  key))
```
```
DuplicateKeyError: E11000 duplicate key error index: clementigroup.rp.session.giotto.rice.edu.eh22.017756.0008.$_id_ dup key: { : "umgr.0000" }
```

**Comment by vivek-bala** [comment]
I agree that you get the following error with devel (0.7.5): 
```
2018-08-13 12:38:22,238: radical.entk.task_processor: task-manager : MainThread : ERROR : CU creation failed, error: attribute set is not extensible/private (key lfs_per_process) (/scratch1/eh22/conda/envs/extasy9/lib/python2.7/site-packages/saga/attributes.py +1068 (_attributes_i_set) : raise se.IncorrectState ("attribute set is not extensible/private (key %s)" % key))
```

Can you confirm that you get the same with latest release/master (0.7.4)? Note that there was release made last week (will send out notifications for future point releases).

**Comment by euhruska** [comment]
hmm, 0.7.4 seems to work

**Comment by vivek-bala** [comment]
Good to hear! Yea, the old devel got merged with master and released as 0.7.4. The new devel has some more features (0.7.5). Apologies on the silent change since there were some API changes that weren't made back compatible.
